### PR TITLE
add an example accordion block style built with vue

### PR DESCRIPTION
**Pull request by oxyc**
This demonstrates how to add block style with a different blade template and a Vue accordion component that's readable by Search Engines and non-js browsers. It's a somewhat advanced example because I also addad ARIA support. Apparently there is a law nowadays that all government funded sites have to be accessible. Luckily both SLS and FSI use foundation which has pretty good accessibility support.

Essentially you can add a single (or multiple) Accordion with the following markup:

```vue 
<accordion>
  <!-- <template> tags are not rendered without JS so this is for SEO -->
  <h3 v-if="false">{!! get_the_title($post) !!}</h3>

  <template #title>{!! get_the_title($post) !!}</template>

  {!! wpautop(get_the_excerpt($post)) !!}
</accordion>
```

To make it linkable:

```vue 
<accordion id="{{ $post->ID }}" :deep-link="true">
  <template #title>{!! get_the_title($post) !!}</template>
  {!! wpautop(get_the_excerpt($post)) !!}
</accordion>
```

To collapse other accordions in a list when one is opened:

```vue 
<accordion-list>
  <accordion></accordion>
  <accordion></accordion>
</accordion-list>
```

With role attributes:

```vue 
<accordion-list role="tablist">
  <accordion role="tab"></accordion>
  <accordion role="tab"></accordion>
</accordion-list>
```

### CODE REVIEW

**Comment by oxyc** [comment]
This brings forth an issue with not having Bootstrap/Foundation. We might have to learn about ARIA roles at least when we work with some clients.

I guess it wouldn't be that many components if we wanted to create them ourselves / use libraries.

**Comment by oxyc** [comment]
Oh and this also works in the Block Editor and basically adds out of the box support for all Vue components created.

Screen Shot 2019-06-11 at 15 55 03

File: resources/assets/scripts/components/accordion.vue [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      <slot></slot>
+    </div>
+  </div>
+</template>
+
+<style>
+</style>
+
+<script>
+import { MixinId } from 'vue-a11y-utils';
```
Comment (oxyc) [inline resources/assets/scripts/components/accordion.vue:16]: This provides `this.localId` which uses `this.id` if available. If not it creates a unique identifier. This is needed for eg. telling screen readers that this specific anchor opens this specific container. If we use `deep-link` option we should set `id` manually as the `localId` gets a new ID every page request.

File: resources/assets/scripts/components/accordion.vue [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      this.active = true;
+      this.$parent.$emit('opened', this);
+
+      if (this.href !== '#') {
+        this.updateHash();
+      }
+    },
+
+    updateHash() {
+      window.history.replaceState({}, '', this.href);
```
Comment (oxyc) [inline resources/assets/scripts/components/accordion.vue:61]: By using `replaceState` we avoid scrolling the browser to the anchor link.

File: resources/assets/scripts/components/accordion.vue [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+      if (accordionList.indexOf(this) !== -1) {
+        this.close();
+      }
+    });
+
+    this.$parent.$on('open', (accordion) => {
+      if (accordion === this) {
+        this.open();
+      }
+    });
```
Comment (oxyc) [inline resources/assets/scripts/components/accordion.vue:83]: Here we listen to any parent element telling us to close or open. Doesn't actually have to be an `<accordion-list>`

**Comment by kimlehtinen** [comment]
Nice

File: resources/assets/scripts/components/accordion.vue (hunk 2 above)
Comment (toffebjorkskog) [reply resources/assets/scripts/components/accordion.vue:61]: i like this

**Pull request closed without merging**